composer/loggers/in_memory_logger.py

```diff
@@ -74,3 +75,50 @@ def log_metric(self, timestamp: Timestamp, log_level: LogLevel, data: TLogData):
             self.data[k].append((timestamp, log_level, v))
         self.most_recent_values.update(data)
         self.most_recent_timestamps.update({k: timestamp for k in data})
+
+    def get_timeseries(self, metric: str, time_unit: Union[str, None] = None):
+        """Returns logged data as dict containing values of a desired metric over time.
+
+        Args:
+            metric (str): Metric of interest. Must be present in self.data.keys().
+            time_unit(str or None, optional) Which unit of time to return. Options =
+                {"batch", "epoch", "sample", "token", "duration", None}. If ``None``,
+                will return all available units of time. Default=``None``.
+
+        Returns:
+            timeseries (dict): Dictionary in which one key is ``metric``, and the
+                associated value is a list of values of that metric. The remaining key(s)
+                are each a unit of time, and the associated values are each a list of
+                values of that time unit for the corresponding index of the metric. For
+                example, plotting timeseries['accuracy/val'] vs. timeseries['epoch'] would
```

Comment:
It does not support returning multiple metrics. Different metrics may have different sampling resolutions and hence different length time series. I think this would require returning nested dicts, which feels like worse UX to me than making multiple calls to `.get_timeseries()`, but others may disagree and also it's possible there's a nice solution I'm not thinking of.

Reply:
Agreed!